super0.c

```diff
@@ -923,12 +926,35 @@ static int store_super0(struct supertype *st, int fd)
 	return 0;
 }
 
+static __u64 avail_size0(struct supertype *st, __u64 devsize,
+			 unsigned long long data_offset);
 static int write_init_super0(struct supertype *st)
 {
 	mdp_super_t *sb = st->sb;
 	int rv = 0;
 	struct devinfo *di;
 
+	if (sb->level == 0 && sb->layout == UnSet) {
+		/* Without requesting a dangerous (0) layout
+		 * we can only allow this RAID0 if all devices are
+		 * the same size
+		 */
+		for (di = st->info; di; di = di->next) {
+			struct devinfo *di2 = st->info;
```

Comment:
What do you think about?
```c
	if (sb->level == 0 && sb->layout == UnSet) {
		/* Without requesting a dangerous (0) layout
		 * we can only allow this RAID0 if all devices are
		 * the same size
		 */
		unsigned long long size = 0;

		for (di = st->info; di; di = di->next) {
			unsigned long long disk_size;

			disk_size = avail_size0(st, di->dev_size, 0) << 9;
			disk_size /= sb->chunk_size;

			if (size == 0) {
				size = disk_size;
				continue;
			}

			if (size != disk_size) {
				pr_err("Need explicit layout=dangerous to create 0.90 raid0 on non-uniform sized devices\n");
				return 1;
			}
		}
```

I wanted to avoid repeated initialization of `s2` and code duplication